### PR TITLE
Add CONTRIBUTING.md

### LINKED ISSUE #26: Create a CONTRIBUTING.md
The main terraform project has notes on how to contribute, and guidance for maintainers, but it's not immediately obvious when looking at this repo in isolation.

Add suitable linking to the README.md, or a new CONTRIBUTING.md, or something. 

https://github.com/hashicorp/terraform/blob/master/docs/maintainer-etiquette.md

### PR DESCRIPTION
**Pull request by armsnyder**
closes #26 

I added a CONTRIBUTING.md with some very basic PR checklist items and guidelines.

Note that the whole "Developing the Provider" section was moved wholesale from the README.md.

Preview: https://github.com/gitlabhq/terraform-provider-gitlab/blob/contributing/CONTRIBUTING.md

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+Generally we accept any change that adds or changes a Terraform resource that is in line with the [GitLab API](https://docs.gitlab.com/ee/api/api_resources.html). It is always best to [open an issue](https://github.com/gitlabhq/terraform-provider-gitlab/issues/new/choose) before starting on a change.
+
+## Getting Started
+
+Use HashiCorp's [Plugin Development](https://www.terraform.io/plugin) guide as a reference, especially the [Provider Design Principles](https://www.terraform.io/plugin/hashicorp-provider-design-principles).
+
+See the [Developing The Provider](#developing-the-provider) section below for specifics about this GitLab provider.
+
+## PR Checklist
```
Comment (timofurrer) [inline CONTRIBUTING.md:13]: Maybe we should also add this into a PR template.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+Terraform resources have a unique ID (`d.SetId("")`). The ID should be comprised from the URL path variables of the `GET` API for the resource in the [GitLab API documentation](https://docs.gitlab.com/ee/api/api_resources.html), separated by `:`.
+
+For example, a resource for the `GET /projects/:id/environments/:environment_id` API would have the ID `123:456` where `123` is the project ID and `456` is the environment ID.
+
+This makes it very easy to add import functionality, by using
+
+```go
+Importer: &schema.ResourceImporter{
+    State: schema.ImportStatePassthrough,
+},
```
Comment (timofurrer) [inline CONTRIBUTING.md:46]:
```suggestion
Importer: &schema.ResourceImporter{
    StateContext: schema.ImportStatePassthroughContext,
},
```

Let's promote the context-aware functions here.

File: CONTRIBUTING.md (hunk 1 above)
Comment (nagyv) [reply CONTRIBUTING.md:13]: Opened https://github.com/gitlabhq/terraform-provider-gitlab/pull/788

**Comment by willianpaixao** [comment]
Great idea @armsnyder,

How about adding a link to the [provider documentation guidelines](https://www.terraform.io/registry/providers/docs)?
Possibly even to the [Doc Preview Tool](https://registry.terraform.io/tools/doc-preview), although is already linked in the page above.

**Comment by armsnyder** [comment]
@willianpaixao Thanks, I added a section for that.

**Review by nagyv — APPROVED** [review]
As Nicholas gave over the project administration to me. I'll approve it.

**Pull request merged**